dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task04330WorkflowTaskAddLanguageIdColumn.java

```diff
@@ -0,0 +1,275 @@
+
+
+package com.dotmarketing.startup.runonce;
+
+import static com.dotcms.util.CollectionsUtils.*;
+
+import com.dotcms.util.CollectionsUtils;
+import com.dotcms.util.ConversionUtils;
+import com.dotmarketing.common.db.DotConnect;
+import com.dotmarketing.db.DbConnectionFactory;
+import com.dotmarketing.db.HibernateUtil;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotHibernateException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.startup.AbstractJDBCStartupTask;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.UtilMethods;
+
+import java.sql.SQLException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This upgrade task will add the column language_id to the workflow_task, set the default language id to all of current records
+ * and add an index for the language_id.
+ *
+ * @author jsanca
+ * @version 5.0
+ *
+ */
+public class Task04330WorkflowTaskAddLanguageIdColumn extends AbstractJDBCStartupTask {
+
+    private static final Map<DbType, String> selectLanguageIdColumnSQLMap   = map(
+            DbType.POSTGRESQL,   "SELECT language_id FROM workflow_task",
+            DbType.MYSQL,        "SELECT language_id FROM workflow_task",
+            DbType.ORACLE,       "SELECT language_id FROM workflow_task",
+            DbType.MSSQL,        "SELECT language_id FROM workflow_task"
+            );
+
+    private static final Map<DbType, String> addLanguageIdColumnSQLMap      = map(
+            DbType.POSTGRESQL,   "ALTER TABLE workflow_task ADD language_id INT8",
+            DbType.MYSQL,        "ALTER TABLE workflow_task ADD language_id bigint",
+            DbType.ORACLE,       "ALTER TABLE workflow_task ADD language_id number(19,0)",
+            DbType.MSSQL,        "ALTER TABLE workflow_task ADD language_id NUMERIC(19,0) null"
+    );
+
+    private static final Map<DbType, String> updateLanguageIdColumnSQLMap   = map(
+            DbType.POSTGRESQL,   "UPDATE workflow_task SET language_id = ?",
+            DbType.MYSQL,        "UPDATE workflow_task SET language_id = ?",
+            DbType.ORACLE,       "UPDATE workflow_task SET language_id = ?",
+            DbType.MSSQL,        "UPDATE workflow_task SET language_id = ?"
+    );
+
+    private static final Map<DbType, String> addLanguageIdIndexSQLMap       = map(
+            DbType.POSTGRESQL,   "create index idx_workflow_6 on workflow_task (language_id)",
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "create index idx_workflow_6 on workflow_task (language_id)" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)